core/src/main/scala/akka/persistence/cassandra/journal/CassandraRecovery.scala

```diff
@@ -4,39 +4,47 @@
 
 package akka.persistence.cassandra.journal
 
-import scala.concurrent._
 import java.lang.{ Long => JLong }
 
+import akka.Done
 import akka.actor.ActorLogging
+import akka.pattern.ask
 import akka.persistence.PersistentRepr
-import akka.persistence.cassandra.journal.TagWriter.{ TagProgress, TagWrites }
-import akka.stream.scaladsl.Source
+import akka.persistence.cassandra.journal.CassandraJournal.{ SequenceNr, Tag }
+import akka.persistence.cassandra.journal.TagWriter.{ SetTagProgress, TagProgress, TagWrite }
+import akka.persistence.cassandra.query.EventsByPersistenceIdStage.TaggedPersistentRepr
+import akka.stream.scaladsl.{ Sink, Source }
+import akka.util.Timeout
 
-import scala.collection.JavaConversions._
+import scala.collection.JavaConverters._
+import scala.concurrent._
+import scala.concurrent.duration._
 
 trait CassandraRecovery extends ActorLogging {
   this: CassandraJournal =>
 
   import config._
   import context.dispatcher
 
+  // TODO this serialises and re-serialises the messages for fixing tag_views
+  // Could have an events by persistenceId stage that has the raw payload
   override def asyncReplayMessages(
```

Comment:
Perhaps we should have a config setting to disable tags completely, and then we could optimize some things, such as the recovery, when tags are disabled. This can be a follow up ticket.

Note that it one can use different plugins with different config, so if an application is using tags for some actors and no tags for others they could use different plugin settings.

Reply:
https://github.com/akka/akka-persistence-cassandra/issues/271